### ISSUE TITLE
Error handling

### ISSUE THREAD

**Issue by ghost** [alexheretic/ktx#12]
Currently the function `ktx::Ktx::new` panics if an invalid file is provided:

```
fn main() {
    unsafe {
        ktx::Ktx::new(std::fs::read("/home/fctorial/.zshrc").unwrap());
    }
}
```

What should I do if I'm trying to load an unknown ktx file, for example, a file that I'm receiving over the network. Is there another function that returns an `std::error::Error` if the provided file is invalid? Does the library currently perform sanity checks on the input file (`pixel_height == 0 && pixel_depth != 0` is invalid,  `numArrayLayers` must be zero for 3d images, cubemaps must be 2d)?

**Comment by alexheretic** [comment]
Yep the library panics on non-ktx files by checking the header. Currently
this is a debug_assert but we could add a `try_new` fn that returns a
Result for that.

**Comment by ghost** [comment]
What about other kind of errors? i.e. Files that have the correct ktx structure but some of the fields contradict each other. If the new function finishes without error, is it guaranteed that the ktx file adhers to the ktx spec?

**Comment by alexheretic** [comment]
I'm less convinced that extra validation should be done on `new`. Maybe if
we had some examples of invalid ktx textures to test with?

But perhaps we can get started with the `try_new` method checking for the
correct ktx header and improve when it looks worthwhile.
